runtime/autoload/provider/ruby.vim

```diff
@@ -7,14 +7,24 @@ let g:loaded_ruby_provider = 1
 let s:stderr = {}
 let s:job_opts = {'rpc': v:true}
 
-function! s:job_opts.on_stderr(chan_id, data, event)
+function! s:job_opts.on_stderr(chan_id, data, event) dict
   let stderr = get(s:stderr, a:chan_id, [''])
   let last = remove(stderr, -1)
   let a:data[0] = last.a:data[0]
-  call extend(stderr, a:data)
+  call extend(stderr, filter(a:data, 'v:val !~# "^Exception ignored"'))
```

Comment:
Since I'm checking the status code before showing the error, the filter is not necessary any more.